website/static/js/share/utils.js

```diff
@@ -221,14 +221,25 @@ utils.maybeQuashEvent = function (event) {
 
 /* Adds a filter to the list of filters if it doesn't already exist */
 utils.updateFilter = function (vm, filter, required) {
-    if (required && vm.requiredFilters.indexOf(filter) === -1) {
-        vm.requiredFilters.push(filter);
-    } else if (vm.optionalFilters.indexOf(filter) === -1 && !required) {
-        vm.optionalFilters.push(filter);
-    }
+    filters = utils.wrapFilter(filter);
+    filters.forEach(function(f){
+        if (required && vm.requiredFilters.indexOf(f) === -1) {
+            vm.requiredFilters.push(f);
+        } else if (vm.optionalFilters.indexOf(f) === -1 && !required) {
+            vm.optionalFilters.push(f);
+        }
+    });
     utils.search(vm);
 };
 
+/* Helper function to accept string or array of strings for updateFilter */
+utils.wrapFilter = function(filter){
```

Comment:
Contrary to what this name suggests, there's nothing really "filter"-specific about this function--this just ensures an Array return value. The name and implementation can be generalized further:

```javascript
function ensureArray(value) {
    if (!Array.isArray(value)) {
        return [value];
    }
    return value;
}
```